aiogram/dispatcher/filters/builtin.py

```diff
@@ -694,14 +695,17 @@ async def check(self, message: Message):
 
 
 class ChatTypeFilter(BoundFilter):
-    key = 'chat_types'
+    key = 'chat_type'
 
-    def __init__(self, chat_types: typing.List[ChatType]):
-        self.chat_types = chat_types
+    def __init__(self, chat_type: typing.Container[ChatType]):
+        if isinstance(chat_type, str):
+            chat_type = {chat_type}
+
+        self.chat_type: typing.Set[str] = set(chat_type)
 
     async def check(self, obj: Union[Message, CallbackQuery]):
         if isinstance(obj, Message):
             obj = obj.chat
         if isinstance(obj, CallbackQuery):
             obj = obj.message.chat
-        return obj.type in self.chat_types
+        return obj.type in self.chat_type
```

Comment:
suggestion if we respect obj delcared types (`Union[Message, CallbackQuery]`):
```python3
    async def check(self, obj: Union[Message, CallbackQuery]):
        if isinstance(obj, Message):
            chat = obj.chat
        elif isinstance(obj, CallbackQuery):
            chat = obj.message.chat
        else:
            return NotImplemented
        return chat.type in self.chat_type
```

Reply:
yeap, sure! thank you
b9424106a71df0857e8456e6b7f77d9b0d6e1401

Reply:
eventually, here is a warning, if input object doens't match the type hint
https://github.com/aiogram/aiogram/pull/356/commits/d98b4ba1811e506e60821de8c316edba65b04ece